backend/internal/role/model.go

```diff
@@ -0,0 +1,198 @@
+type RoleSummaryResponse struct {
+	ID                 string `json:"id"`
+	Name               string `json:"name"`
+	Description        string `json:"description,omitempty"`
+	OrganizationUnitID string `json:"ouId"`
+}
+
+// RoleResponse represents a complete role with permissions.
+type RoleResponse struct {
+	ID                 string   `json:"id"`
+	Name               string   `json:"name"`
+	Description        string   `json:"description,omitempty"`
+	OrganizationUnitID string   `json:"ouId"`
+	Permissions        []string `json:"permissions"`
+}
+
+// CreateRoleRequest represents the request body for creating a role.
+type CreateRoleRequest struct {
+	Name               string              `json:"name"`
+	Description        string              `json:"description,omitempty"`
+	OrganizationUnitID string              `json:"ouId"`
+	Permissions        []string            `json:"permissions"`
+	Assignments        []AssignmentRequest `json:"assignments,omitempty"`
+}
+
+// CreateRoleResponse represents the response body for creating a role.
+type CreateRoleResponse struct {
+	ID                 string               `json:"id"`
+	Name               string               `json:"name"`
+	Description        string               `json:"description,omitempty"`
+	OrganizationUnitID string               `json:"ouId"`
+	Permissions        []string             `json:"permissions"`
+	Assignments        []AssignmentResponse `json:"assignments,omitempty"`
+}
+
+// UpdateRoleRequest represents the request body for updating a role.
+type UpdateRoleRequest struct {
+	Name               string   `json:"name"`
+	Description        string   `json:"description,omitempty"`
+	OrganizationUnitID string   `json:"ouId"`
+	Permissions        []string `json:"permissions"`
+}
+
+// AssignmentsRequest represents the request body for adding or removing assignments.
+type AssignmentsRequest struct {
+	Assignments []AssignmentRequest `json:"assignments"`
+}
+
+// LinkResponse represents a pagination link.
+type LinkResponse struct {
+	Href string `json:"href"`
+	Rel  string `json:"rel"`
+}
+
+// RoleListResponse represents the response for listing roles with pagination.
+type RoleListResponse struct {
+	TotalResults int                   `json:"totalResults"`
+	StartIndex   int                   `json:"startIndex"`
+	Count        int                   `json:"count"`
+	Roles        []RoleSummaryResponse `json:"roles"`
+	Links        []LinkResponse        `json:"links"`
+}
+
+// AssignmentListResponse represents the response for listing role assignments with pagination.
+type AssignmentListResponse struct {
+	TotalResults int                  `json:"totalResults"`
+	StartIndex   int                  `json:"startIndex"`
+	Count        int                  `json:"count"`
+	Assignments  []AssignmentResponse `json:"assignments"`
+	Links        []LinkResponse       `json:"links"`
+}
+
+// Internal service layer structs - used for business logic processing
```

Comment:
```suggestion
```

We don't need these comments right?

Reply:
I would prefer to keep it as a separator to differentiate the structs used in the handler and service layer